x-pack/plugins/cloud/public/components/chat/iframe.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useRef, useState, CSSProperties } from 'react';
+import { css } from '@emotion/react';
+import { useChat } from '../../services';
+
+type UseChatType =
+  | { enabled: false }
+  | {
+      enabled: true;
+      src: string;
+      ref: React.MutableRefObject<HTMLIFrameElement | null>;
+      style: CSSProperties;
+    };
+
+const MESSAGE_READY = 'driftIframeReady';
+const MESSAGE_RESIZE = 'driftIframeResize';
+const MESSAGE_SET_CONTEXT = 'driftSetContext';
+
+const iframeStyle = css`
+  position: fixed;
+  botton: 30px;
+  right: 30px;
+  display: block;
+`;
+
+// We're sending a lot of information to the frame, so this method puts together the specific
+// properties, 1/ to avoid leaking too much, and 2/ to enumerate precisely what we're sending.
+const getContext = () => {
+  const { location, navigator, innerHeight, innerWidth } = window;
+  const { hash, host, hostname, href, origin, pathname, port, protocol, search } = location;
+  const { language, userAgent } = navigator;
+  const { title, referrer } = document;
+
+  return {
+    window: {
+      location: {
+        hash,
+        host,
+        hostname,
+        href,
+        origin,
+        pathname,
+        port,
+        protocol,
+        search,
+      },
+      navigator: { language, userAgent },
+      innerHeight,
+      innerWidth,
+    },
+    document: {
+      title,
+      referrer,
+    },
+  };
+};
+
+const useFrame = (): UseChatType => {
+  const ref = useRef<HTMLIFrameElement>(null);
+  const chat = useChat();
```

Comment:
NIT/optional: I would rename the hook `useChatConfig`, and potentially also rename the property name on the `CloudService` type from `chat` to `chatConfig` (at least if we're not expecting to add any function to the return of the hook).